students/hyunsoo-jeon/user/views.py

```diff
@@ -0,0 +1,42 @@
+import json, re
+
+from django.views             import View
+from django.http              import JsonResponse
+from django.core.exceptions   import ValidationError
+
+from .models                  import User
+
+
+class SignupView(View):
+    def post(self, request):
+        data      = json.loads(request.body)
+        email_reg = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+        regex     = re.compile(email_reg)
+
+        try:
+            if not regex.match(data['email']):
+                return JsonResponse ({"message": "This email is not valid"}, status=400)
+
+            if len(data['password']) < 8:
+                 return JsonResponse ({"message": "This password is too short"}, status=400)
+
+            if User.objects.filter(email = data['email']).exists():
+                return JsonResponse ({"message": "This email already exists."}, status=400)
+
+            if data.get('nickname') != None and User.objects.filter(nickname = data['nickname']).exists():
```

Comment:
```suggestion
            if data.get('nickname') is None and User.objects.filter(nickname = data['nickname']).exists():
```
None을 비교 할 때는 is을 주로 사용합니다.